packages/core/src/actions.ts

```diff
@@ -159,7 +159,7 @@ export function toActivityDefinition<TContext, TEvent extends EventObject>(
  */
 export function raise<TContext, TEvent extends EventObject>(
   event: Event<TEvent>
-): RaiseAction<TEvent> | SendAction<TContext, TEvent, TEvent> {
+): RaiseAction<TEvent> | SendAction<TContext, AnyEventObject, TEvent> {
```

Comment:
Confused about this - the `raise(...)` action creator should only ever be able to raise events that the machine accepts; that is, only `TEvent` events, not `AnyEventObject`.

Reply:
Right, the second type param is about the event received by send expressions - `raise` doesn't support expressions right now though, so we can just parameterized this with `AnyEventObject` so it can be passed at any position in a machine's config. The 3rd type param in `send` is `TSentEvent` and this contraints the `raise` to only raise events accepted by the machine.

Reply:
Ah, gotcha. Sounds good to me.